File: 程式設計：資料結構答案檔/Python/104.py
import sys

def find_lower_area(map, x, y, w, h):
    if map[x][y] > 0:
        return 0
    if map[x][y] == -2:
        return 0
    if map[x][y] == -1:
        return 0
    map[x][y] = -1
    return (1 + find_lower_area(map, x-1, y, w, h) + find_lower_area(map, x+1, y, w, h)
              + find_lower_area(map, x, y-1, w, h) + find_lower_area(map, x, y+1, w, h))

def show_map(map, w, h):
    for i in range(1, w+1):
        for j in range(1, h+1):
            print(map[i][j], end=" ")
        print()
    print()

m, n = (int(x) for x in input().split())

map = [[-2 for _ in range(n+2)] for _ in range(m+2)] 

for i in range(1, m+1):
    row = [int(x) for x in input().split()]
    for j in range(1, n+1):
        map[i][j] = row[j-1]

maxSize = 0
for i in range(1, m+1):
    for j in range(1, n+1):
        size = find_lower_area(map, i, j, m, n)
        if size > maxSize:
            maxSize = size

print(maxSize)

File: 程式設計：資料結構答案檔/C#/104.cs
using System;

class Program {
    static int FindLowerArea(int[,] map, int x, int y, int w, int h) {
        if (map[x,y] > 0 || map[x,y] == -2 || map[x,y] == -1) {
            return 0;
        }
        map[x,y] = -1;
        return 1 + FindLowerArea(map, x-1, y, w, h) + FindLowerArea(map, x+1, y, w, h)
                 + FindLowerArea(map, x, y-1, w, h) + FindLowerArea(map, x, y+1, w, h);
    }

    static void Main() {
        string[] inputs = Console.ReadLine().Split();
        int m = int.Parse(inputs[0]);
        int n = int.Parse(inputs[1]);

        int[,] map = new int[m+2, n+2];
        for (int i = 0; i < m+2; i++) {
            for (int j = 0; j < n+2; j++) {
                map[i,j] = -2;
            }
        }

        for (int i = 1; i <= m; i++) {
            inputs = Console.ReadLine().Split();
            for (int j = 1; j <= n; j++) {
                map[i,j] = int.Parse(inputs[j-1]);
            }
        }

        int maxSize = 0;
        for (int i = 1; i <= m; i++) {
            for (int j = 1; j <= n; j++) {
                int size = FindLowerArea(map, i, j, m, n);
                if (size > maxSize) {
                    maxSize = size;
                }
            }
        }
        Console.WriteLine(maxSize);
    }
}
